### PR TITLE
[Wasm] Rename System.Runtime.InteropServices.JavaScript to System.Private.Runtime.InteropServices.JavaScript

### LINKED ISSUE #39042: S.R.InteropServices.JavaScript is part of the shared framework but doesn't have a ref assembly
System.Runtime.InteropServices.JavaScript is currently declared as part of the shared framework but doesn't have a reference assembly. As the project isn't OOB either, there is currently no way for a customer to reference the library besides directly referencing the implementation one from the shared framework directory.

Is the omission of a reference assembly intentional?

cc @kjpou1 @marek-safar @ericstj

### PR DESCRIPTION
**Pull request by steveisok**
Fixes https://github.com/dotnet/runtime/issues/39042

### CODE REVIEW

File: src/mono/wasm/runtime/binding_support.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
 var BindingSupportLib = {
 	$BINDING__postset: 'BINDING.export_functions (Module);',
 	$BINDING: {
-		BINDING_ASM: "[System.Runtime.InteropServices.JavaScript]System.Runtime.InteropServices.JavaScript.Runtime",
+		BINDING_ASM: "[System.Private.Runtime.InteropServices.JavaScript]System.Runtime.InteropServices.JavaScript.Runtime",
 		mono_wasm_object_registry: [],
```
Comment (lewing) [inline src/mono/wasm/runtime/binding_support.js:8]: There appears to be some fallback code(??) a few lines down in the file that names the assembly directly.

**Review by safern — APPROVED** [review]
LGTM

File: src/mono/wasm/runtime/binding_support.js (hunk 1 above)
Comment (steveisok) [reply src/mono/wasm/runtime/binding_support.js:8]: Do you mean here?  https://github.com/dotnet/runtime/blob/427f4f09cebdefc768909461af8f08625c2fb3c2/src/mono/wasm/runtime/binding_support.js#L78

Unless I'm mistaken, that's using the namespace, which is still the same.

Comment (lewing) [reply src/mono/wasm/runtime/binding_support.js:8]: You are correct!

**Review by lewing — APPROVED** [review]
Looks good

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**